app/models/message.rb

```diff
@@ -2,13 +2,20 @@ class Message < ActiveRecord::Base
   belongs_to :user
 
   validates_presence_of :to_phone_number, :body
+  validate :user_quota_limit
+
 
   after_commit :send_sms_job
 
   attr_reader :group_id
 
+  def user_quota_limit
+    self.user.quota_limit_check
+    errors.add(:message, "would exceed your quota. So it cannot be sent.")
```

Comment:
You're correct @mattruzicka I fixed it when I realized it. Was fixed in 2fd1b73b72012477b4c8305a31dafc8c1b78a65b, by wrapping it in an unless. May not be the cleanest way to do this, but my motto is that development is a process. Have to keep iterating.